frontend/src/components/Base.js

```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import {connect} from "react-redux";
+import Home from "./Home";
+import Result from "./Result";
+
+
+/* 
+  Classe base da aplicação, 
+  utilizada para carregar a interface inicial e de resultado
+  da pesquisa realizada
+*/
+
+class Base extends React.Component {
```

Comment:
Quando vi o nome desse component, que se chama Base, me levou acreditar que ele seria um componente de base para todos os outros que existirem na aplicação, mas não parece ser o caso ( podemos ver o que voce pensou quando criou ele ). 

Segundo o **Clean Code** nomes precisam ter/mostrar a suas intenções e se faz necessário evitar nomes muito genéricos: 

_**A long descriptive name is better than a short enigmatic name. A long descriptive name is better than a long descriptive comment... Robert C. Martin**_

Acredito que poderíamos incluir ao nome ```Base``` algumas informações para dar uma especialização nele, como por exemplo: 

```
class BaseHomeInitalScreen extends React.Component
//OR
class HomeBaseScreenSelector extends React.Component
//OR
class HomeScreenSelector extends React.Component
// OR
class HomeScreenRouter extends React.Component
// OR
class HomeScreenStateSelector extends React.Component 
```

Nesse caso acredito que o ultimo nome seria umas das opções ou um caminho a seguir para escolher o nome da classe, pois vejo que no método render ele está mudando o estado da ```Home``` para escolher a tela(Screen) que ele irá dar ```display```, então o nome ```HomeScreenStateSelector``` ou algo parecido seria um bom caminha a se tomar.